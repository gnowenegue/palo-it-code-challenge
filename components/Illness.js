import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import AppContext from '../contexts/AppContext';

const Illness = ({ illness: { id, name } }) => {
  const { setIllness } = useContext(AppContext);

  const updateIllness = e => {
    e.preventDefault();

    document.querySelectorAll('a.collection-item').forEach(elem => {
      elem.classList.remove('active');
      elem.classList.add('blue-text');
    });

    e.currentTarget.classList.toggle('active');
    e.currentTarget.classList.toggle('blue-text');
    setIllness({
      id: e.currentTarget.dataset.illnessid,
      name: e.currentTarget.dataset.illness,
    });
  };

  return (
    <a
      href="#!"
      data-target="severity"
      data-illnessid={id}
      data-illness={name}
      onClick={updateIllness}
      className="collection-item blue-text waves-effect waves-red modal-trigger"
    >
      {name}
      <i className="secondary-content material-icons">send</i>

      <style jsx>
        {`
          a.collection-item:not(.active):hover {
            background-color: #ffcdd2;
          }
          a.collection-item.active {
            background-color: #b71c1c;
            color: white;
          }
        `}
      </style>
    </a>
  );
};

Illness.propTypes = {
  illness: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default Illness;
