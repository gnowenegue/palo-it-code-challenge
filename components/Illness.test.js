import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';

import AppContext from '../contexts/AppContext';
import Illness from './Illness';

describe('<Illness />', () => {
  test('Snapshot test', () => {
    const render = renderer.create(
      <AppContext.Provider value={{ setIllness: () => {} }}>
        <Illness illness={{ id: 1, name: 'flu' }} />
      </AppContext.Provider>
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
