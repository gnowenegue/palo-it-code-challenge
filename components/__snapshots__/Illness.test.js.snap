// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Illness /> Snapshot test 1`] = `
<a
  className="jsx-4047661875 collection-item blue-text waves-effect waves-red modal-trigger"
  data-illness="flu"
  data-illnessid={1}
  data-target="severity"
  href="#!"
  onClick={[Function]}
>
  flu
  <i
    className="jsx-4047661875 secondary-content material-icons"
  >
    send
  </i>
</a>
`;
